### PR TITLE
Controllers

### PR DESCRIPTION
**Pull request by brahimdidi**
In this part I created controllers and view for all my models

## Requirments
For each URL you should create:

A route.
An action in the correct controller.
A view file.
View files should include only placeholders written in plain HTML, e.g:

Here is a list of posts for a given user
Make sure that you added code to handle only the 4 required URLs.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 Rails.application.routes.draw do
+  resources :users, shallow: true do
+    resources :comments
+    resources :posts
```
Comment (mikethreels) [inline config/routes.rb]: - [ ] try to avoid unnecessarily exposing end-points

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Didi, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (brahimdidi) [reply config/routes.rb]: No clue

Comment (brahimdidi) [reply config/routes.rb]: If I remove shallow nothing works, thanks.

**Review by nicupop729 — COMMENTED** [review]
Hello @brahimdidi 👋,

_This is an additional review as a result of the second opinion request._

Nice implementation of the project requirements! :clap:

What the CR meant is that you should only use the `index` and `show` (later on you will need the `create` and `new` also) and not all routes, as they are not needed. 
Please check [this link for reference](https://guides.rubyonrails.org/routing.html#configuring-the-rails-router)!

Please see if these settings will help you:
`resources :users, only: [:index, :show] do`
    --`resources :posts, only: [:index, :show]`
 `end`

@mikethreels, when requesting a change, please explain why the change is requested, and please make sure to be understood by the student. Also, please reply in the thread.

@brahimdidi, after you implement the changes, please ask again for another review!

Good luck and keep rocking 🚀!

**Comment by mikethreels** [comment]
@brahimdidi, I'm sorry I was not able to respond earlier, I would have explained the requirements in more detail.

@nicupop729, thank you for the explanation I will keep in mind to be more detailed when requiring changes.

**Comment by brahimdidi** [comment]
Thank you @nicupop729 , no worries @mikethreels

**Review by Munsa1 — APPROVED** [review]
Hi @brahimdidi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**